### PR TITLE
Add optional vsock support

### PR DESCRIPTION
**Pull request by guillaumerose**

### CODE REVIEW

**Comment by praveenkumar** [comment]
@guillaumerose shouldn't we also update the circle ci for running the tests and linter?

**Comment by guillaumerose** [comment]
I added it in the Dockerfile.

File: Makefile [hunk 1]
```diff
@@ -35,3 +35,5 @@ release: build
 	    echo '```' ; \
 	done
 
+lint:
+	golangci-lint run
```
Comment (cfergeau) [inline Makefile]: github UI complains this does not end witha newline

File: libvirt.go [hunk 2]
```diff
@@ -257,13 +258,17 @@ func (d *Driver) Create() error {
...
-		DiskPath:   d.DiskPath,
-		Network:    d.Network,
+		DomainName:   d.MachineName,
+		Memory:       d.Memory,
+		CPU:          d.CPU,
+		CacheMode:    d.CacheMode,
+		IOMode:       d.IOMode,
+		DiskPath:     d.DiskPath,
+		Network:      d.Network,
+		ExtraDevices: []string{},
```
Comment (cfergeau) [inline libvirt.go]: Isn't this going to be empty by default if you omit it?

File: Makefile [hunk 3]
```diff
@@ -35,8 +34,13 @@ release: build
 	    echo '```' ; \
 	done
 
+.PHONY: validate
+validate: test lint
+
+.PHONY: test
```
Comment (cfergeau) [inline Makefile:40]: This `.PHONY test` should go with the test target which was added in one of the previous commits

**Review by cfergeau — APPROVED** [review]
A few minor comments, but /lgtm

File: Makefile (hunk 1 above)
Comment (guillaumerose) [reply Makefile]: done :+1:

File: libvirt.go (hunk 2 above)
Comment (guillaumerose) [reply libvirt.go]: done :+1:

File: Makefile (hunk 3 above)
Comment (guillaumerose) [reply Makefile:40]: done :+1:

File: constants.go [hunk 4]
```diff
@@ -2,7 +2,7 @@ package libvirt
 
 const (
 	DriverName    = "libvirt"
-	DriverVersion = "0.12.9"
+	DriverVersion = "0.12.8"
```
Comment (praveenkumar) [inline constants.go]: @guillaumerose why this change?

Comment (guillaumerose) [reply constants.go]: this is a mistake, sorry!
